services/github/github-tag.service.js

```diff
@@ -73,8 +83,8 @@ class GithubTag extends GithubAuthV4Service {
     }
   }
 
-  async fetch({ user, repo, sort }) {
-    const limit = sort === 'semver' ? 100 : 1
+  async fetch({ user, repo, sort, prefix }) {
+    const limit = sort === 'semver' || prefix ? 100 : 1
```

Comment:
I can't add the comment down below but, is there any reason we couldn't incorporate the prefix (when one is provided) as part of the query itself?

Reply:
I tried to update the graphql but a prefix is not supported. I can add a `query` parameter but that matches by the [github query syntax ](https://docs.github.com/en/search-github/getting-started-with-searching-on-github/understanding-the-search-syntax) which might be a nice way to search but prefix is not supported there either.

Reply:
I haven't fully read this PR, but we recently added another badge which uses the graphql API to query tags by prefix using `refPrefix`:

https://github.com/badges/shields/blob/37a36f5feac3c2143ea06fb9b6528ff5979f2dd2/services/ros/ros-version.service.js#L62-L82

does that help?

Reply:
@chris48s thanks for reviewing!

When working on this change I tried editing the `refPrefix` line and I don't know what is accepted there. Putting a prefix of the tag as that `${distro}` variable errors and doesn't return anything. Maybe I am just using it wrong but I wasn't able to find documentation on what would go there.

Also, I don't see any variable in the shield.io UI that allows a prefix when you search for `github tag` under Versions. 

If this is already done I would be happy to just use that I just didn't see a way.

Reply:
Tangentially related, but you may find GitHub's Explorer useful in the future when fiddling around with their GraphQL queries: https://docs.github.com/en/graphql/overview/explorer

What I was alluding to, and I believe Chris as well, is that it seems like we'd be able to utilize the value of the `prefix` parameter of this function directly as part of the GraphQL query we are issuing which would allow GitHub to do the work for us.

Specifically, I think this would entail adding a param on line 80/90 (old/new line):

```js
       query ($user: String!, $repo: String!, $limit: Int!, $prefix: String!) {
```

and then incorporate that param down on line 83/93:
```js
refPrefix: "refs/tags/$prefix"
```

finally updating the variables on line 96/106

```js
variables: { user, repo, limit, prefix },
```

It's not clear to me what issue you were encountering or if this was even something you'd already tried, but even if the above doesn't work literally as-is I'd still think something like this would be possible. E.g. if the string interpolation for the ref prefix) then we could set the refPrefix value outside the query and just pass that value in as the variable (`refPrefix: $prefix`)

Reply:
@calebcartwright thanks for sharing that explorer, it was super helpful! I guess it comes down to what you want the prefix to do. The  `refPrefix` requires that a tag have a prefix that ends with `/`

I want this prefix for Golang sub packages and their requirement is to have a `folder/` prefix but the ticket that I used for this PR wanted to have any generic prefix.

For example my package `mvndaai/ctxerrhelper` currently has 4 tags `["logrus/v0.0.2", "logrus/v0.0.1", "example/v0.0.2",  "example/v0.0.1"]`
If I search for the prefix `example/` then I get back ~`["v0.0.2", "v0.0.1"]`. The only allowed prefixes are `example` and `logrus` which totally works for me but doesn't work for the other generic ask of something like `ex`.

Here is an example query that you can use if you want to test
```graphql
query {
    repository(owner: "mvndaai", name: "ctxerrhelper") {
        refs(
            refPrefix: "refs/tags/example/"
            first: 30
            orderBy: { field: TAG_COMMIT_DATE, direction: DESC }
        ) {
            edges { node { name } }
        }
    }
}
```
--------
I guess the questions are:
- Is it okay to require a prefix to end with `/`?
- Since only tags uses graphql, are you happy if tags and releases work differently?

Reply:
I decided to simplify this PR only change only tags, no releases, and use the graphql refPrefix.

Reply:
I had not realised that the `refPrefix` param requires the prefix to end with a slash. Sorry if I've muddied the waters there. I can see it will be useful to filter by any prefix. For example, if you think about the tags on this repo https://github.com/badges/shields/tags?after=server-2021-07-01 I can see you might want to filter by `server-`. Maybe the original implementation would be better after all. Sorry if I've sent us on a goose chase here.

Reply:
I think both ways have their place. Maybe I will update this tonight to handle both ways with a different param for the prefix that requires a '/'